### PR TITLE
Unhook the mouse down hook at the end of PropertyGridView.ProcessEnumUpAndDown

### LINKED ISSUE #12434: [Accessibility] Pressing Up/Down keyboard in drop-down list of property collapses drop-down directly
### .NET version

10.0.100-alpha.1.24553.6

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, all versions have this issue

### Issue description

Pressing Up/Down keyboard in drop-down list of property collapses dropdown directly.
Image

**Expand Result**: Drop-down list keeps expanding state when pressing Up/Down keyboard, then pressing Enter can collapse it

### Steps to reproduce

Repro steps:
1.	Find one property with dropdown list in PropertyGrid, for example: **Multiline**
2.	Tab to its value part: **False**, then press Up keyboard  to change value: **True** --- it is necessary step
3.	Press Alt+Down or F4 to expand this property's drop-down list
4.	Press Down keyboard to choose another value: **False**

### PR DESCRIPTION
**Pull request by LeafShi1**
Fixes #12434

## Root cause:

After changing the value by pressing up/down keyboard directly in EditTextbox, the `EditTextBox.HookMouseDown` will be set to True

The stack trace:
```
at System.Windows.Forms.PropertyGridInternal.PropertyGridView.GridViewTextBox.set_HookMouseDown(Boolean value)
at System.Windows.Forms.PropertyGridInternal.PropertyGridView.SetCommitError(ErrorState error, Boolean capture)
at System.Windows.Forms.PropertyGridInternal.PropertyGridView.OnEditChange(Object sender, EventArgs e)
at System.Windows.Forms.Control.OnTextChanged(EventArgs e)
at System.Windows.Forms.TextBoxBase.OnTextChanged(EventArgs e)
at System.Windows.Forms.Control.set_Text(String value)
at System.Windows.Forms.TextBoxBase.set_Text(String value)
at System.Windows.Forms.TextBox.set_Text(String value)
at System.Windows.Forms.PropertyGridInternal.PropertyGridView.GridViewTextBox.set_Text(String value)
at System.Windows.Forms.PropertyGridInternal.PropertyGridView.CommitValue(GridEntry entry, Object value, Boolean closeDropDown)
at System.Windows.Forms.PropertyGridInternal.PropertyGridView.ProcessEnumUpAndDown(GridEntry entry, Keys keyCode, Boolean closeDropDown)


```
then pressing `Alt+Down` to open the DropDown list, and select a item, the [CommitValue ](https://github.com/dotnet/winforms/blob/7896a6e7b4d3627c76bd3168d0042f72574d9b14/src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/PropertyGridView.cs#L4663) will be invoked. 
At this time, `EditTextBox.HookMouseDown = true`. In [HookMouseDown](https://github.com/dotnet/winforms/blob/7896a6e7b4d3627c76bd3168d0042f72574d9b14/src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/PropertyGridView.GridViewTextBox.cs#L64-L68), the `Focus()` was invoked, so the DropDown list was collapsed.

## Proposed changes

-  Set  `EditTextBox.HookMouseDown = false` at the end of PropertyGridView.ProcessEnumUpAndDown

## Customer Impact

- When pressing the up/down keyboard in the property drop-down list, the drop-down menu will not be collapsed directly

## Regression? 

- No

## Risk

- Mininal

### Before
Pressing Up/Down keyboard in drop-down list of property collapses dropdown directly.
Image

### After
When pressing the up/down keyboard in the property drop-down list, the drop-down menu will not be collapsed directly
AfterChange

## Test methodology 

- Manually

## Test environment(s) 

- .net 10.0.0-beta.24562.15

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12508)

### CODE REVIEW

**Comment by ricardobossan** [comment]
I proceeded to test the changes made in this PR and the problem seems to be solved:

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/PropertyGridView.cs [hunk 1]
```diff
@@ -2475,6 +2475,7 @@ private bool ProcessEnumUpAndDown(GridEntry entry, Keys keyCode, bool closeDropD
             }
 
             CommitValue(entry, valueNew, closeDropDown);
+            EditTextBox.HookMouseDown = false;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/PropertyGridView.cs:2478]: Should it be 
```suggestion
            EditTextBox?.HookMouseDown = false;
```

**Comment by LeafShi1** [comment]
No new issue was found with the private dlls

**Review by Tanya-Solyanik — APPROVED** [review]
Looks good

**Pull request merged**